emails/views.py

```diff
@@ -381,6 +387,14 @@ def _sns_message(message_json):
     address.save(
         update_fields=['num_forwarded', 'last_used_at']
     )
+
+    # only remove message from S3 if the email was stored in S3
+    if 'receipt' in message_json and 'action' in message_json['receipt']:
+        if 'S3' in message_json['receipt']['action']['type']:
+            bucket = message_json['receipt']['action']['bucketName']
+            object_key = message_json['receipt']['action']['objectKey']
+            remove_message_from_s3(bucket, object_key)
```

Comment:
I was going to say we should handle failures to remove messages from S3, but it looks like `remove_message_from_s3` does enough error logging that we can detect problems and clean out the s3 bucket if we need to do it manually later?

Reply:
Yup. Since `remove_message_from_s3` runs in the background I wanted it to fail more silently. So we are logging it on Sentry AND on the metrics so ops and engineers can be more aware of it when the failure to delete the message object happens.